stdlib/cmake/modules/AddSwiftStdlib.cmake

```diff
@@ -799,11 +799,11 @@ function(add_swift_target_library_single target name)
         "${SWIFT_SDK_MACCATALYST_LIB_SUBDIR}/${SWIFTLIB_SINGLE_ARCHITECTURE}")
   endif()
 
-  if ("${SWIFTLIB_SINGLE_BOOTSTRAPPING}" STREQUAL "")
+  if ("${SWIFTLIB_SINGLE_BOOTSTRAPPING}" STREQUAL "" OR NOT SWIFTLIB_SINGLE_SDK IN_LIST SWIFT_DARWIN_PLATFORMS)
     set(output_sub_dir ${SWIFTLIB_SINGLE_SUBDIR})
   else()
     # In the bootstrapping builds, we only have the single host architecture.
-    # So generated the library directly in the parent SDK specific directory
+    # So generate the Darwin library directly in the parent SDK specific directory
```

Comment:
It may be possible to continue placing the bootstrapping libraries in a non-architecture-specific directory like on Darwin, then only link the rpath for the final bootstrapped Swift compiler against the architecture-specific directory, but I'm not interested in digging into all the CMake config to make that happen.

It is easier to simply place the bootstrapping libraries in architecture-specific directories too on ELF platforns. I don't check for Windows here because my understanding is that bootstrapping the compiler doesn't work on Windows: @compnerd, correct me if I'm wrong.

Reply:
Nope, bootstrapping isn't enabled yet on Windows.